app.js

```diff
@@ -75,24 +77,37 @@ app.use(session({
 app.use('/_api', authRoutes);
 
 ////////
-// LOGIN
+// AUTHENTICATION
 ////////
 
-if (config.get('AUTH_ENABLED')) {
-    // if username and password supplied, enable auth
-    app.use('/login', loginRoutes);
-    app.use('/logout', logoutRoutes);
+if(config.get('AUTH_STRATEGY')) {
+    // Initialize passport and restore auth state from session
+    app.use(passport.initialize());
+    app.use(passport.session());
 
-    //routes registered below this filter will require a valid session value/user
-    app.all('*',function(req,res,next) {
-        if(req.session.username) {
-            next();
-        }
-        else {
-            //save the previous page in the session to know where to redirect back to after login
-            req.session.redirectAfterLogin = req.path;
-            res.redirect('/login');
-        }
+    // Load auth strategy based on config
+    var strategy = require(config.get('AUTH_STRATEGY'))(app);
+    passport.use(strategy);
```

Comment:
A non-existent module crashes the app immediately.

`make dev AUTH_STRATEGY=./app/qwerqwer`

A module that doesn't return a strategy crashes the app immediately.

`make dev AUTH_STRATEGY=./app/auth-bogus`

implemented as:

```
module.exports = function() {
    return {};
};
```